code/game/objects/items/stacks/flags.dm

```diff
@@ -76,3 +76,138 @@
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	item_icons = list(
+		WEAR_L_HAND = 'icons/mob/humans/onmob/items_lefthand_64.dmi',
+		WEAR_R_HAND = 'icons/mob/humans/onmob/items_righthand_64.dmi'
+		)
+
+	var/flag_type = /obj/structure/flag/plantable /// The typepath of the flag structure that gets spawned when the flag is planted.
+	var/faction /// Used to check if nearby mobs belong to a faction when calculating for the stronger warcry.
+	var/play_warcry = FALSE /// Does the flag play a unique warcry when planted? (Only while on harm intent.)
+	var/warcry_sound /// The warcry's sound path.
+	var/warcry_extra_sound /// When there are more than 14 allies nearby, play this stronger warcry.
+	COOLDOWN_DECLARE(warcry_cooldown_item) /// Used to limit the spam of the warcry_extra_sound
+
+/obj/item/flag/plantable/get_examine_text()
+	. = ..()
+	if(play_warcry)
+		. += SPAN_NOTICE("Planting the flag while in <b>HARM</b> intent will cause you to bellow out a rallying warcry!")
+
+/obj/item/flag/plantable/proc/plant_flag(mob/living/user, play_warcry = FALSE, warcry_sound, warcry_extra_sound, faction)
+	if(user.action_busy)
+		return
+
+	if(SSinterior.in_interior(user))
+		to_chat(usr, SPAN_WARNING("There's no way to plant [src] in here!"))
+		return
+
+	var/turf/turf_to_plant = get_step(user, user.dir)
+	for(var/obj/object in turf_to_plant)
+		if(object.density)
+			to_chat(usr, SPAN_WARNING("You need a clear, open area to plant [src], something is blocking the way in front of you!"))
+			return
+
+	user.visible_message(SPAN_NOTICE("[user] starts planting [src] into the ground..."), SPAN_NOTICE("You start planting [src] into the ground..."))
+	playsound(user, 'sound/effects/flag_raising.ogg', 30)
+	if(!do_after(user, 6 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+		return
+
+	user.visible_message(SPAN_NOTICE("[user] plants [src] into the ground!"), SPAN_NOTICE("You plant [src] into the ground!"))
+	var/obj/structure/flag/plantable/planted_flag = new flag_type(turf_to_plant)
+
+	// If there are more than 14 allies nearby, play a stronger rallying cry.
+	// Otherwise, play the default warcry sound if there is one. If not, play a generic flag raising sfx.
+	if(play_warcry && user.faction == faction && user.a_intent == INTENT_HARM)
+		var/allies_nearby = 0
+		if(COOLDOWN_FINISHED(src, warcry_cooldown_item))
+			for (var/mob/living/carbon/human in orange(planted_flag, 7))
+				if (human.stat == DEAD)
+					continue
+				if (human.faction != faction)
+					continue
+				allies_nearby++
+				if (prob(40) && human != user)
+					human.emote("warcry")
+
+		user.show_speech_bubble("warcry")
+		if(allies_nearby > 14)
+			playsound(user, warcry_extra_sound, 30)
+			// Start a cooldown on the flag structure. This way we can keep track of the cooldown when the flag is hoisted and taken down.
+			COOLDOWN_START(planted_flag, warcry_cooldown_struc, 90 SECONDS)
+			user.say("*me shouts an invigorating rallying cry!")
```

Comment:
you could probably use the emote proc instead

Reply:
the only gripe i have with the emote proc is that there doesn't seem to be a way to force runechat on it. it only displays in the chatbox for me which is kind of underwhelming

Reply:
i think i found more suitable proc than say() now, works with runechat too. should be resolved i guess